src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/resource/mapper/induction/InPrisonInterestsResourceMapper.kt

```diff
@@ -0,0 +1,25 @@
+package uk.gov.justice.digital.hmpps.educationandworkplanapi.app.resource.mapper.induction
+
+import org.mapstruct.Mapper
+import org.mapstruct.NullValueMappingStrategy
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.domain.induction.dto.CreateInPrisonInterestsDto
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.resource.model.CreateInPrisonInterestsRequest
+
+@Mapper(nullValueIterableMappingStrategy = NullValueMappingStrategy.RETURN_DEFAULT)
+abstract class InPrisonInterestsResourceMapper {
+  fun toCreateInPrisonInterestsDto(
+    request: CreateInPrisonInterestsRequest?,
+    prisonId: String,
+  ): CreateInPrisonInterestsDto? {
+    return if (request == null) {
```

Comment:
If it wasn't for this null check, all the "resource" mappers would be a single line interface (which goes to show how great `MapStruct` is when the mapping is simple).

Reply:
Remind me again why we need to do these null checks?

Reply:
Because the generated MapStruct method uses an and (`&&`) rather than an or (`||`):

```
if ( request == null && prisonId == null ) {
   return null;
}
```

Reply:
Ah, thats right, I remember now 👍

Reply:
It's a known problem and obviously not easy to fix (given this discussion is over 2 years old): https://github.com/mapstruct/mapstruct/discussions/2642

Reply:
As discussed on Slack, we now have a much better solution which is to avoid passing nullable source parameters into the mappers in the first place (by using `.let` in the main `InductionResourceMapper`).